### PR TITLE
feat(cli): enable CLI args for lhci wizard new-project

### PR DESCRIPTION
**Pull request by revolunet**
Hi,

This makes it possible to run the `new-project` command without user interaction.

One could create a project like this :

```sh
lhci wizard --wizard=new-project \
  --basicAuth.username=admin \
  --basicAuth.password=io \
  --serverBaseUrl="https://lighthouse.hello.world" \
  --projectName="some-new-project" \
  --projectExternalUrl="https://somewhere.else" \
  --projectBaseBranch="master"

```

### CODE REVIEW

**Comment by revolunet** [comment]
@googlebot I fixed it.

File: packages/cli/src/wizard/wizard.js [hunk 1]
```diff
@@ -29,33 +29,46 @@ async function runNewProjectWizard(options) {
   const fallbackProjectName = path.basename(process.cwd());
   const [defaultProjectName = fallbackProjectName] = (repoSlug && repoSlug.match(/[^/]+$/)) || [];
 
-  const responses = await inquirer.prompt([
-    {
+  const prompts = [];
```
Comment (patrickhulce) [inline packages/cli/src/wizard/wizard.js:32]: You'll want to add a typedef here, I'm not sure exactly what the type will be but it'll look something like this


```suggestion
  /** @type {Array<import('inquirer').WhateverThePromptTypeIs>} */
  const prompts = [];
```

File: packages/cli/src/wizard/wizard.js [hunk 2]
```diff
@@ -29,33 +29,46 @@ async function runNewProjectWizard(options) {
...
       message: "What branch is considered the repo's trunk or main branch?",
       default: 'master',
-    },
-  ]);
+    });
+  }
+
+  const responses = {
+    ...options,
+    ...(await inquirer.prompt(prompts)),
```
Comment (patrickhulce) [inline packages/cli/src/wizard/wizard.js:70]: awesome! could we get a test case added? :) we should have some good examples of invoking the CLI already in there. maybe copy `healtcheck` to a new `wizard-no-prompts` one if you don't see a good existing place to add?

**Review by patrickhulce — CHANGES_REQUESTED** [review]
thanks very much for the contribution @revolunet! great thinking!

File: packages/cli/src/wizard/wizard.js (hunk 1 above)
Comment (revolunet) [reply packages/cli/src/wizard/wizard.js:32]: sorry, TS noob here, how to find the correct type ?

Comment (patrickhulce) [reply packages/cli/src/wizard/wizard.js:32]: if you're using vscode you can ctrl/cmd+click on the `import('inquirer')` bit and poke around the type definitions. In this case we want the same thing that `prompt` accepts, which looks like `Array<import('inquirer').Question>`

File: packages/cli/test/wizard.test.js [hunk 3]
```diff
@@ -74,6 +74,34 @@ describe('wizard CLI', () => {
...
+          '--wizard=new-project',
+          '--basicAuth.username=admin',
+          '--basicAuth.password=io',
+          '--serverBaseUrl="https://lighthouse.hello.world"',
+          '--projectName="some-new-project"',
+          '--projectExternalUrl="https://somewhere.else"',
+          '--projectBaseBranch="master"',
+        ],
+        [
+          // '', // Just ENTER key to select "new-project"
```
Comment (patrickhulce) [inline packages/cli/test/wizard.test.js:92]: remove these?

File: packages/cli/test/wizard.test.js [hunk 4]
```diff
@@ -74,6 +74,34 @@ describe('wizard CLI', () => {
...
+    it('should create a new project with arguments', async () => {
+      server.process.kill();
+      server = await startServer(server.sqlFile, ['--basicAuth.password=lighthouse']);
+
+      const {stdout, stderr, status} = await runWizardCLI(
+        [
+          '--wizard=new-project',
+          '--basicAuth.username=admin',
+          '--basicAuth.password=io',
+          '--serverBaseUrl="https://lighthouse.hello.world"',
```
Comment (patrickhulce) [inline packages/cli/test/wizard.test.js:86]: this is going to have to match the server you started

File: packages/cli/test/wizard.test.js [hunk 5]
```diff
@@ -74,6 +74,34 @@ describe('wizard CLI', () => {
...
 
+    it('should create a new project with arguments', async () => {
+      server.process.kill();
+      server = await startServer(server.sqlFile, ['--basicAuth.password=lighthouse']);
+
+      const {stdout, stderr, status} = await runWizardCLI(
+        [
+          '--wizard=new-project',
+          '--basicAuth.username=admin',
+          '--basicAuth.password=io',
```
Comment (patrickhulce) [inline packages/cli/test/wizard.test.js:85]: same here though I don't think we need basicAuth as part of this test

**Review by patrickhulce — COMMENTED** [review]
tried to give some pointers on the failing test :)

**Comment by revolunet** [comment]
i cant figure out the issues on the typecheck


```
packages/cli/src/wizard/wizard.js:81:5 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

81     rootURL: responses.serverBaseUrl || options.serverBaseUrl,
       ~~~~~~~

  packages/utils/src/api-client.js:17:15
    17    * @param {{rootURL: string, fetch?: import('isomorphic-fetch'), URL?: typeof import('url').URL, extraHeaders?: Record<string, string>, basicAuth?: LHCI.ServerCommand.Options['basicAuth']}} options
                     ~~~~~~~
    The expected type comes from property 'rootURL' which is declared here on type '{ rootURL: string; fetch?: ((input: RequestInfo, init?: RequestInit | undefined) => Promise<Response>) | undefined; URL?: typeof URL | undefined; extraHeaders?: Record<...> | undefined; basicAuth?: { ...; } | undefined; }'

packages/cli/src/wizard/wizard.js:84:5 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

84     name: responses.projectName,
       ~~~~

  types/server.d.ts:20:9
    20         name: string;
               ~~~~
    The expected type comes from property 'name' which is declared here on type 'Pick<Project, "slug" | "createdAt" | "updatedAt" | "name" | "externalUrl" | "baseBranch">'

packages/cli/src/wizard/wizard.js:85:5 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

85     externalUrl: responses.projectExternalUrl,
       ~~~~~~~~~~~

  types/server.d.ts:21:9
    21         externalUrl: string;
               ~~~~~~~~~~~
    The expected type comes from property 'externalUrl' which is declared here on type 'Pick<Project, "slug" | "createdAt" | "updatedAt" | "name" | "externalUrl" | "baseBranch">'

packages/cli/src/wizard/wizard.js:86:5 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

86     baseBranch: responses.projectBaseBranch,
       ~~~~~~~~~~

  types/server.d.ts:23:9
    23         baseBranch: string;
               ~~~~~~~~~~
    The expected type comes from property 'baseBranch' which is declared here on type 'Pick<Project, "slug" | "createdAt" | "updatedAt" | "name" | "externalUrl" | "baseBranch">'

```

**Comment by patrickhulce** [comment]
For the typecheck failures there are a couple options with our new dynamic sourcing but the most straightforward that should work for this situation is assert that they're defined after our prompt.

```js
if (!responses.serverBaseUrl) throw new Error('Failed to respond with a server URL');
if (!responses.projectName) throw new Error('Failed to respond with a project name');
```

If that doesn't work I've got a backup just so we can merge and I'll take a look at them later.

**Pull request closed without merging**